### PR TITLE
Add Option to Export CSV

### PR DESCRIPTION
**Pull request by gitttt-1234**
### Description
Add an option to export the analysis file in CSV format

### Types of changes

- [ ] Bugfix
- [ ] New feature
- [ ] Refactor / Code style update (no logical changes)
- [ ] Build / CI changes
- [ ] Documentation Update
- [ ] Other (explain)

### Does this address any currently open issues?
[list open issues here]

### Outside contributors checklist

- [ ] Review the [guidelines for contributing](https://github.com/talmolab/sleap/blob/develop/docs/CONTRIBUTING.md) to this repository
- [ ] Read and sign the [CLA](https://github.com/talmolab/sleap/blob/develop/sleap-cla.pdf) and add yourself to the [authors list](https://github.com/talmolab/sleap/blob/develop/AUTHORS)
- [ ] Make sure you are making a pull request against the **develop** branch (not *main*). Also you should start *your branch* off *develop*
- [ ] Add tests that prove your fix is effective or that your feature works
- [ ] Add necessary documentation (if appropriate)

#### Thank you for contributing to SLEAP!
:heart:

### CODE REVIEW

**Review by roomrys — COMMENTED** [review]
Possible to combine `ExportAnaliysisFile` and `ExportCSVFile`?

File: sleap/gui/commands.py [hunk 1]
```diff
@@ -1122,6 +1127,108 @@ def ask(context: CommandContext, params: dict) -> bool:
         params["filename"] = filename
         return True
 
+class ExportCSVFile(AppCommand):
```
[the remark this answers is not in the document]
Comment (gitttt-1234) [reply sleap/gui/commands.py:1130]: Hi,
yes it was almost the same. I changed the code and combined the classes, will push the changes in my next commit

File: tests/io/test_formats.py [hunk 2]
```diff
@@ -125,6 +126,21 @@ def test_hdf5_v1_filehandle(centered_pair_predictions_hdf5_path):
         == "tests/data/json_format_v1/centered_pair_low_quality.mp4"
     )
 
+def test_csv(tmpdir, centered_pair_predictions):
+    from sleap.info.write_tracking_h5 import main as write_analysis
+
+    filename = os.path.join(tmpdir, "analysis.csv")
```
Comment (roomrys) [inline tests/io/test_formats.py:132]: Prefer pathlib

File: tests/io/test_formats.py [hunk 3]
```diff
@@ -125,6 +126,21 @@ def test_hdf5_v1_filehandle(centered_pair_predictions_hdf5_path):
...
+    video = centered_pair_predictions.videos[0]
+
+    write_analysis(centered_pair_predictions, output_path=filename, all_frames=True, csv=True)
+
+    labels = pd.read_csv(
+        filename, header=True
+    )
+
+    # TODO: assert 
+
```
Comment (roomrys) [inline tests/io/test_formats.py:142]: Add test fixture artifact with correct CSV for comparison

**Review by roomrys — CHANGES_REQUESTED** [review]

File: sleap/info/write_tracking_h5.py [hunk 4]
```diff
@@ -286,12 +287,77 @@ def write_occupancy_file(
...
+        print(f"No tracks to export in {data_dict['video_path']}. Skipping the export")
+        return
+
+    data_dict["node_names"] = [s.decode() for s in data_dict["node_names"]]
+    data_dict["track_names"] = [s.decode() for s in data_dict["track_names"]]
+    data_dict["track_occupancy"] = np.transpose(data_dict["track_occupancy"]).astype(
+        bool
+    )
+
+    # find frames with at least one animal tracked
```
Comment (roomrys) [inline sleap/info/write_tracking_h5.py:313]: Nit: google style comments
```suggestion
    # Find frames with at least one animal tracked.
```

File: sleap/io/format/csv.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
+"""
+Adaptor for writing SLEAP analysis as csv.
+
+"""
```
Comment (roomrys) [inline sleap/io/format/csv.py:4]: Nit: google style docstrings

```suggestion
"""Adaptor for writing SLEAP analysis as csv."""
```

File: sleap/io/format/csv.py [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+from sleap import Labels, Video
+from typing import Optional, Callable, List, Text, Union
+
+
+class CSVAdaptor(format.adaptor.Adaptor):
+    FORMAT_ID = 1.2
+
+    # 1.0 points with gridline coordinates, top left corner at (0, 0)
+    # 1.1 points with midpixel coordinates, top left corner at (-0.5, -0.5)
+    # 1.2 adds track score to read and write functions
```
Comment (roomrys) [inline sleap/io/format/csv.py:17]:
```suggestion
    FORMAT_ID = 1.0

    # 1.0 initial implementation
```

File: sleap/info/write_tracking_h5.py [hunk 7]
```diff
@@ -1,4 +1,4 @@
-"""Generate an HDF5 file with track occupancy and point location data.
+"""Generate an HDF5/ CSV file with track occupancy and point location data.
```
Comment (roomrys) [inline sleap/info/write_tracking_h5.py:1]:
```suggestion
"""Generate an HDF5 or CSV file with track occupancy and point location data.
```

**Review by roomrys — COMMENTED** [review]
Good job on condensing the classes. 

I feel like we can condense even further though and use some of the variable for user selection downstream in the FileDialog... ie. instead of having a two buttons for Export Analysis CSV and Export Analysis HDF5, we have one button Export Analysis and allow the user to select the export format as csv downstream in the FileDialog.... The only trouble would be specifying the default extension to use when exporting analysis for all files (and only selecting a folder via the FileDialog...).

What do you think?

**Review by roomrys — APPROVED** [review]

**Pull request merged**